website/src/auth.tsx

```diff
@@ -1,53 +1,132 @@
-      Auth: {
-        // TODO How to get these variables from the server?
-        region: process.env["DAILP_AWS_REGION"],
-        userPoolId: process.env["DAILP_USER_POOL"],
-        userPoolWebClientId: process.env["DAILP_USER_POOL_CLIENT"],
-      },
-    })
+  const [user, setUser] = useState<CognitoUser | null>(
+    userPool.getCurrentUser()
+  )
 
-    Auth.currentUserPoolUser()
-      .then((user) => setUser(user))
-      .catch(() => setUser(null))
-
-    const listener: HubCallback = async (data) => {
-      switch (data.payload.event) {
-        case "signIn":
-        case "signUp":
-          setUser(await Auth.currentUserPoolUser())
-          break
-        case "signOut":
-          setUser(null)
-          break
-      }
+  // refreshes last authenticated user
+  useEffect(() => {
+    if (user != null) {
+      user.getSession(function (err: Error, result: CognitoUserSession | null) {
+        if (result) {
+          console.log("You are now logged in.")
+        }
+      })
     }
-    Hub.listen("auth", listener)
-    return () => Hub.remove("auth", listener)
   }, [])
 
+  // React.useEffect(() => {
+  // Amplify.configure({
+  //   Auth: {
+  //     // TODO How to get these variables from the server?
+  //     region: process.env["DAILP_AWS_REGION"],
+  //     userPoolId: process.env["DAILP_USER_POOL"],
+  //     userPoolWebClientId: process.env["DAILP_USER_POOL_CLIENT"],
+  //   },
+  // })
+
+  // Auth.currentUserPoolUser()
+  //   .then((user) => setUser(user))
+  //   .catch(() => setUser(null))
+
+  //   const listener: HubCallback = async (data) => {
+  //     switch (data.payload.event) {
+  //       case "signIn":
+  //         setUser(userPool.getCurrentUser());
+  //       case "signUp":
+  //       // setUser(await Auth.currentUserPoolUser())
+  //       // break
+  //       case "signOut":
+  //         setUser(null)
+  //         break
+  //     }
+  //   }
+  //   Hub.listen("auth", listener)
+  //   return () => Hub.remove("auth", listener)
+  // }, [])
+
+  function loginUser(username: string, password: string) {
+    const user = new CognitoUser({
+      Username: username,
+      Pool: userPool,
+    })
+
+    const authDetails = new AuthenticationDetails({
+      Username: username,
+      Password: password,
+    })
+
+    // logs in the user with the authentication details
+    user.authenticateUser(authDetails, {
+      onSuccess: (data: CognitoUserSession) => {
+        console.log("Login success. Result: ", data)
+        alert("Login successful")
+        setUser(user)
+      },
+      onFailure: (data: CognitoUserSession) => {
+        console.error("Login failed. Result: ", data)
+        alert("Login failed")
+      },
+      newPasswordRequired: (data: CognitoUserSession) => {
+        console.log("New password required. Result: ", data)
+        alert("New password is required")
+      },
+    })
+  }
+
+  function logoutUser(user: CognitoUser) {
+    user?.signOut()
```

Comment:
Would it make sense to just use the `user` object already in scope instead of needing a function argument?

Reply:
That does make sense - let me go and change that.